api/cpp/CMakeLists.txt

```diff
@@ -439,10 +439,17 @@ set(CPACK_RESOURCE_FILE_README "${CMAKE_CURRENT_LIST_DIR}/README.md")
 set(CPACK_STRIP_FILES ON)
 set(CPACK_NSIS_DEFINES "ManifestDPIAware true")
 
-if(NOT WIN32)
+if(WIN32)
+    if(MSVC)
+        set(compiler_suffix "MSVC")
+    elseif(MINGW)
+        set(compiler_suffix "MinGW")
+    endif()
+    set(CPACK_SYSTEM_NAME "${CPACK_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}-${CPACK_SYSTEM_NAME}-${compiler_suffix}")
```

Comment:
If it is neither msvc nor mingw, the system will end with a dash.
I don't think we support anything else, but just to be more robust, it might make sense to put the dash in the compiler_suffix